### PR TITLE
Add Top-k sensitivity metrics + ROC & Precision-Recall curves

### PR DESCRIPTION
**Pull request by adriaromero**
Fix [ch591] [ch588] used for the Q-Sub performance evaluation. 

## Top-k sensitivity:
- Retrieve an old `get_top_k_sensitivity` function (I removed in the past)
- Add in `utils.py` check functions: `check_input_samples` and `check_top_k_concepts`
- Added test coverage

## ROC and Precision-Recall curves
- Add both functions in `visualizer.py`: `plot_ROC_curve` and `plot_precision_recall_curve`.
- **IMPORTANT**: I implemented both functions considering only binary classification problems. 
- Added some examples in `notebook.ipynb`
- Added test coverage

### CODE REVIEW

**Comment by tjarmain** [comment]
@adriaromero can you send me an updated evaluation when it's ready for the N way we are featuring in the FDA submission? I have to make sure that all of the investor presentations/materials match these figures. Thanks!

**Comment by adriaromero** [comment]
@tjarmain these are the global and class results: https://gist.github.com/adriaromero/72ff1476e9f41619335af4726085cccf

And the pretty confusion matrix:
screen shot 2018-10-29 at 12 41 39 pm

Let me know if you need the Risk Benchmark study evaluation as well!

File: keras_eval/metrics.py [hunk 1]
```diff
@@ -332,3 +333,44 @@ def get_top1_entropy_stats(y_probs, labels, entropy, verbose=0):
     n_errors = np.array(n_errors)
 
     return correct_list, errors_list, n_correct, n_errors
+
+
+def get_top_k_sensitivity(concepts, y_true, y_probs, top_k, round_decimals=7):
```
Comment (jsalbert) [inline keras_eval/metrics.py]: Why is this computed outside the `get_metrics()` function?
Half of the code is the same isn't it?

File: keras_eval/visualizer.py [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
 
     Args:
-        cm: Confusion Matrix, square sized numpy array
-        concepts: Name of the categories to show
-        normalize: If True, normalize values between 0 and ones. Not valid if negative values.
-        fontsize: Text size
-        figsize: Figure size
-        cmap: Colormap of your choice
-        save_path: If `save_path` specified save confusion matrix in that location
+        cm: confusion matrix in the form of numpy array
```
Comment (jsalbert) [inline keras_eval/visualizer.py]: I was writing every sentence starting in capital letters

File: keras_eval/visualizer.py [hunk 3]
```diff
@@ -1,21 +1,26 @@
...
-def plot_confusion_matrix(cm, concepts, normalize=False, fontsize=18, figsize=(16, 12),
+def plot_confusion_matrix(cm, concepts, normalize=False, show_text=True, fontsize=18, figsize=(16, 12),
                           cmap=plt.cm.coolwarm_r, save_path=None):
     '''
+    Plot confusion matrix provided in 'cm'
 
     Args:
-        cm: Confusion Matrix, square sized numpy array
-        concepts: Name of the categories to show
-        normalize: If True, normalize values between 0 and ones. Not valid if negative values.
```
Comment (jsalbert) [inline keras_eval/visualizer.py]: Normalize is missing now

File: keras_eval/metrics.py (hunk 1 above)
Comment (adriaromero) [reply keras_eval/metrics.py]: I created a new function to break down the `get_metrics` function a bit, but I agree that follows the same structure. I removed `get_top_k_sensitivity ` and added the code back to `get_metrics`.

**Pull request merged**